src/main/java/com/ciandt/techgallery/service/impl/EmailServiceImpl.java

```diff
@@ -123,27 +129,68 @@ public void execute(String userId, String technologyId, String recommendationsId
       String[] recommendIds = recommendationsIds.split(",");
       for (String id : recommendIds) {
         if (!id.isEmpty()) {
-          TechnologyRecommendation recommedation =
-              technologyRecommendationDao.findById(Long.parseLong(id));
+          TechnologyRecommendation recommedation = technologyRecommendationDao.findById(Long.parseLong(id));
           recommendations.add(recommedation);
           comments.remove(recommedation.getCommentEntity());
         }
       }
     }
-    TechGalleryUser user = TechGalleryUserDAOImpl.getInstance().findById(Long.parseLong(userId));
-    Technology technology = TechnologyDAOImpl.getInstance().findById(technologyId);
+    TechGalleryUser user = techGalleryUserDao.findById(Long.parseLong(userId));
+    Technology technology = technologyDao.findById(technologyId);
+
+    TechnologyActivitiesTO techActivities = createEmailTO(user, null, technology, comments, recommendations, new Date(),
+        Constants.APP_NAME);
+
+    EmailConfig email = new EmailConfig(
+        EmailTypeEnum.DAILY_RESUME_MAIL.getSubject() + technology.getName() + " - "
+            + techActivities.getFormattedTimestamp(),
+        "emailtemplates" + File.separator + EmailTypeEnum.DAILY_RESUME_MAIL.getTemplate(), techActivities, null,
+        EmailTypeEnum.DAILY_RESUME_MAIL.getReason(), user.getEmail());
+    sendEmail(email);
+  }
 
+  private TechnologyActivitiesTO createEmailTO(TechGalleryUser user, TechGalleryUser endorserUser,
+      Technology technology, List<TechnologyComment> comments, List<TechnologyRecommendation> recommendations,
+      Date timestamp, String appName) {
     TechnologyActivitiesTO techActivities = new TechnologyActivitiesTO();
+    techActivities.setUser(user);
     techActivities.setComments(comments);
     techActivities.setRecommendations(recommendations);
     techActivities.setTechnology(technology);
     techActivities.setTimestamp(new Date());
-    techActivities.setAppName("Tech Gallery");
+    techActivities.setAppName(appName);
+    techActivities.setEndorserUser(endorserUser);
+    techActivities.setTechnologyLink(generateTechnologyLink(technology));
+    return techActivities;
+  }
+
+  private String generateTechnologyLink(Technology tech) {
+    String linkTechnology;
+    String queryString = "?id=" + tech.getId();
+    String environment = System.getProperty(RUNTIME_ENVIRONMENT_PROPERTY);
+    if (StringUtils.equals(PRODUCTION_PROPERTY, environment)) {
+      String applicationId = System.getProperty(APPLICATION_ID_PROPERTY);
+      String version = System.getProperty(APPLICATION_VERSION_PROPERTY);
+      String versionName = version.split("\\.")[0];
+      linkTechnology = "https://" + versionName + "-dot-" + applicationId + ".appspot.com/";
```

Comment:
@tsr-thulio This could generate an error when we are on production. Please use TechGalleryUtil.getAppId()